app/models/company.rb

```diff
@@ -68,6 +68,20 @@ def location_info(location = locations.first)
     end
   end
 
+  def attach_info(industries, state, location_params)
+    assign_industries(industries)
+    location = locations.create(location_params)
```

Comment:
a helper method like `assign_industries` for location, `attach_location`, would be good here too.

Then this method becomes:

```ruby
def attach_info(industries, state, location_params)
  assign_industries(industries)
  attach_location(state, location_params)
```